pkg/disk/error.go

```diff
@@ -0,0 +1,41 @@
+package disk
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"syscall"
+)
+
+type ErrorMessage struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+func parseErrorMessage(errStr string) (*ErrorMessage, error) {
+	r := regexp.MustCompile(`"code": (-?\d+),\n\t"message": "([^"]+)"`)
+	match := r.FindStringSubmatch(errStr)
+	if len(match) == 0 {
+		return nil, fmt.Errorf("failed to parse error message")
+	}
+
+	code := 0
+	if _, err := fmt.Sscanf(match[1], "%d", &code); err != nil {
+		return nil, fmt.Errorf("failed to parse error code: %w", err)
+	}
+
+	em := &ErrorMessage{
+		Code:    code,
+		Message: match[2],
+	}
+
+	return em, nil
+}
+
+func isFileExists(message string) bool {
```

Comment:
Quesion: Should we make some spdk APIs idempodent? For example, DiskCreate, DiskDelete, Expose...

Reply:
Can you elaborate more on `idempodent` ?

Reply:
Deleting a non-existing disk always returns `nil` rather than an error.
Stopping exposing a stopped replica Lvol always returns `nil`.